src/main/scala/chisel3/util/BitPat.scala

```diff
@@ -108,6 +108,13 @@ sealed class BitPat(val value: BigInt, val mask: BigInt, width: Int) extends Sou
   def === (that: UInt): Bool = macro SourceInfoTransform.thatArg
   def =/= (that: UInt): Bool = macro SourceInfoTransform.thatArg
 
+  override def equals(obj: Any): Boolean = {
+    obj match {
+      case y: BitPat => value == y.value && mask == y.mask && getWidth == y.getWidth
+      case _ => throw new RuntimeException("BitPat can only compare to another BitPat.")
```

Comment:
This should just return false, can consider using the runtime deprecated mechanism to print a warning at the end of Chisel elab.